squad-create-skipfile-reproducers

```diff
@@ -0,0 +1,237 @@
+        )
+
+
+def parse_args(raw_args):
+    parser = argparse.ArgumentParser(
+        description="Produce TuxRun or TuxPlan reproducers for the LTP skipfile."
+    )
+
+    parser.add_argument(
+        "--group",
+        required=True,
+        help="The name of the SQUAD group.",
+    )
+
+    parser.add_argument(
+        "--allow-unfinished",
+        required=False,
+        action="store_true",
+        default=False,
+        help="Allow fetching of reproducers where the build is marked as unfinished.",
+    )
+
+    parser.add_argument(
+        "--branches",
+        required=False,
+        default=supported_branch_names(),
+        nargs="+",
+        help="A list of SQUAD branches to be tested.",
+    )
+
+    parser.add_argument(
+        "--build-names",
+        required=False,
+        default=["gcc-12-lkftconfig", "gcc-\d\d-lkftconfig"],
+        nargs="+",
+        help="The list of accepted build names (for example, gcc-12-lkftconfig). Regex is supported.",
+    )
+
+    parser.add_argument(
+        "--debug",
+        required=False,
+        action="store_true",
+        default=False,
+        help="Display debug messages.",
+    )
+
+    parser.add_argument(
+        "--device-names",
+        required=False,
+        default=["qemu-armv7", "qemu-arm64", "qemu-i386", "qemu-x86_64"],
+        nargs="+",
+        help="The list of device names (for example, qemu-arm64).",
+    )
+
+    parser.add_argument(
+        "--search-build-count",
+        required=False,
+        default=10,
+        type=int,
+        help="The number of builds to fetch when searching for a reproducer.",
+    )
+
+    parser.add_argument(
+        "--skipfile-url",
+        required=False,
+        default="https://raw.githubusercontent.com/Linaro/test-definitions/master/automated/linux/ltp/skipfile-lkft.yaml",
+        help="URL of the skipfile to test.",
+    )
+
+    parser.add_argument(
+        "--suite-name",
+        required=False,
+        default="ltp-syscalls",
+        help="The suite name to grab a reproducer for.",
+    )
+
+    return parser.parse_args(raw_args)
+
+
+def run(raw_args=None):
+    start = time.time()
+    args = parse_args(raw_args)
+
+    skipfile = squadutilslib.get_file(args.skipfile_url)
+
+    reason_list = []
+    reproducer_scripts = []
+
+    with open(skipfile) as f:
+        reasons = yaml.load(f, Loader=yaml.FullLoader)
+    for reason in reasons["skiplist"]:
+        if reason["boards"] == "all" or "all" in reason["boards"]:
+            if reason["branches"] == "all" or "all" in reason["branches"]:
+                projects = [
+                    look_up_project_by_branch(branch) for branch in args.branches
+                ]
+            else:
+                for branch_name in reason["branches"]:
+                    if branch_name in args.branches:
+                        projects.append(look_up_project_by_branch(branch_name))
+                    else:
+                        logger.debug(
```

Comment:
Can't this be done for every branch and we can drop the "else:" bit?

Reply:
Not sure I understand what you mean here - are you talking about the debug print that prints when a branch from the skipfile entry is skipped?